src/uct/cuda/cuda_ipc/cuda_ipc_iface.c

```diff
@@ -74,56 +101,67 @@ static ucs_status_t uct_cuda_ipc_iface_get_address(uct_iface_h tl_iface,
     return UCS_OK;
 }
 
-static int uct_cuda_ipc_iface_is_mnnvl_supported(uct_cuda_ipc_md_t *md)
+static int
+uct_cuda_ipc_iface_mnnvl_reachable(uct_cuda_ipc_md_t *md,
+                                   const uct_cuda_ipc_device_addr_t *dev_addr,
+                                   size_t dev_addr_len,
+                                   const uct_iface_is_reachable_params_t *params)
 {
-#if HAVE_CUDA_FABRIC
-    CUdevice cu_device;
-    int coherent;
-    ucs_status_t status;
-
-    status = UCT_CUDADRV_FUNC_LOG_ERR(cuDeviceGet(&cu_device, 0));
-    if (status != UCS_OK) {
+#if HAVE_NVML_FABRIC_INFO
+    if (memcmp(dev_addr->mnnvl_addr.cluster_uuid, md->fabric_info.clusterUuid,
+               sizeof(dev_addr->mnnvl_addr.cluster_uuid))) {
+        uct_iface_fill_info_str_buf(params, "cluster uuid doesn't match");
         return 0;
     }
 
-    status = UCT_CUDADRV_FUNC_LOG_ERR(
-            cuDeviceGetAttribute(&coherent,
-                                 CU_DEVICE_ATTRIBUTE_PAGEABLE_MEMORY_ACCESS_USES_HOST_PAGE_TABLES,
-                                 cu_device));
-    if (status != UCS_OK) {
+    if (dev_addr->mnnvl_addr.clique_id != md->fabric_info.cliqueId){
+        uct_iface_fill_info_str_buf(params, "clique id doesn't match");
```

Comment:
@brminich The main concern with reachability check I have here is that device 0 determines reachability for the eventual device that this thread will use during transfer/progress time. Devices on the same node aren't required to be part of the same cluster or clique even though that is most likely true.

Reply:
what is your suggestion?